billing-report-agent/src/main/java/com/epam/pipeline/billingreportagent/service/impl/converter/StorageToBillingRequestConverter.java

```diff
@@ -97,10 +102,13 @@ public StorageToBillingRequestConverter(final AbstractEntityMapper<StorageBillin
                                                          final LocalDateTime syncStart) {
         final Long storageId = storageContainer.getEntity().getId();
         final DataStorageType storageType = storageContainer.getEntity().getType();
-        final LocalDate reportDate = syncStart.toLocalDate().minusDays(1);
-        final String fullIndex = indexPrefix + parseDateToString(reportDate);
         return requestSumAggregationForStorage(storageId, storageType)
-            .map(searchResponse -> buildRequestFromAggregation(storageContainer, syncStart, searchResponse, fullIndex))
+            .map(searchResponse -> Stream.iterate(previousSync.plusDays(1L), date -> date.plusDays(1))
```

Comment:
Could you please make this behaviour configurable, by default we create documents only for current date, and optionally for the period